src/common/xml_parser.F

```diff
@@ -1,49 +1,54 @@
-! xmlparse.f90 - Simple, limited XML parser in Fortran
-!
-! $Id: xmlparse.f90,v 1.14 2007/12/07 10:10:19 arjenmarkus Exp $
```

Comment:
please list Arjen as a copyright holder here instead of moving his name to just authors

Reply:
https://sourceforge.net/projects/xml-fortran/ lists the BSD license on the project, so we MUST maintain a copyright notice.

Reply:
I don't want to change the top banner, because the copyright of today's version is shared among several people.

So how about an explicit statement in the Doxgen box:
```
!> \note  Arjen Markus hold the copyright for the original version of this file?
```

Reply:
> I don't want to change the top banner, because the copyright of today's version is shared among several people.

https://reuse.software/faq/#copy-work

Reply:
So, the solution is simply to add for the files in question the proper copyright headers as well as the original licenses. Since they are compatible, the resulting license of a built cp2k executable will still be GPL2-or-later. Then we also don't have to ask the people to relicense (which as @vondele pointed out in earlier discussions is problematic).
The good thing is that now we're using a standardized SPDX header it should be easy to write a generic file header validator.

Reply:
the FSFE even provides a [copyright linter tool](https://pypi.org/project/reuse/), together with a [pre-commit hook](https://github.com/fsfe/reuse-tool/blob/master/.pre-commit-hooks.yaml)

Reply:
> the FSFE even provides a copyright linter tool, together with a pre-commit hook

The point of this PR is to simplify the situation. Hence, I don't want to build out the infrastructure to support even more license mixups.

I found that `xml_parser.F` is completely unused and `glob_matching.F` is barley used. So, can just remove those two.